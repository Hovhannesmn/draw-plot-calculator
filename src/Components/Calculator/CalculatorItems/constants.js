import {
  EQ,
  X_ADD_X,
  X_SUB_X,
  X_MUL_X,
  X_DIV_X,
  ADD,
  SUB,
  DIV,
  MUL,
  CLEAR,
} from '../../../utils/defines';

const INPUT_DEFINES = [
  {
    value: '1',
  },
  {
    value: '2',
  },
  {
    value: '3',
  },
  {
    value: ADD,
    className: 'operator',
  },
  {
    value: '4',
  },
  {
    value: '5',
  },
  {
    value: '6',
  },
  {
    value: SUB,
    className: 'operator',
  },
  {
    value: '7',
  },
  {
    value: '8',
  },
  {
    value: '9',
  },
  {
    value: MUL,
    className: 'operator',

  },
  {
    value: X_ADD_X
  },
  {
    value: X_SUB_X
  },
  {
    value: X_MUL_X
  },
  {
    value: X_DIV_X
  },
  {
    value: CLEAR,
    className: 'clear',
  },
  {
    value: '0',
  },
  {
    value: EQ,
    className: 'equal',
  },
  {
    value: DIV,
    className: 'operator',
  },
];

export {
  INPUT_DEFINES,
};
