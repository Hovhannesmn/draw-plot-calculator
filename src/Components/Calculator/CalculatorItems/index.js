import React from 'react';
import PropTypes from 'prop-types';

import { INPUT_DEFINES } from './constants';
import noop from '../../../utils/noop';

const CalculatorItems = ({
  loading,
  handleChange,
}) => (
  INPUT_DEFINES.map(({ value, className }, index) => (
    <>
      <input
        key={value}
        type="button"
        value={value}
        className={className}
        disabled={loading}
        onClick={() => handleChange(value)}
      />
      {index && index % 3 === 0 ? <br/> : null}
    </>
  ))
);

CalculatorItems.propTypes = {
  loading: PropTypes.bool,
  handleChange: PropTypes.func.isRequired,
};

CalculatorItems.defaultProps = {
  loading: false,
  handleChange: noop,
};

export default CalculatorItems;
