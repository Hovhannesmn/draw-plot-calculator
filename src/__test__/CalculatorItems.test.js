import React from 'react';
import { shallow } from 'enzyme';

import CalculatorItems from '../Components/Calculator/CalculatorItems';

import noop from '../utils/noop';

describe("Test CalculatorItems Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CalculatorItems handleChange={noop} />);
  });

  it('it should render CalculatorItems snapshot when loaded', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
