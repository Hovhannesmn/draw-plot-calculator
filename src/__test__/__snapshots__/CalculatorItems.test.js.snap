// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CalculatorItems Component it should render CalculatorItems snapshot when loaded 1`] = `
Array [
  <Fragment>
    <input
      disabled={false}
      key="1"
      onClick={[Function]}
      type="button"
      value="1"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="2"
      onClick={[Function]}
      type="button"
      value="2"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="3"
      onClick={[Function]}
      type="button"
      value="3"
    />
  </Fragment>,
  <Fragment>
    <input
      className="operator"
      disabled={false}
      key=" + "
      onClick={[Function]}
      type="button"
      value=" + "
    />
    <br />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="4"
      onClick={[Function]}
      type="button"
      value="4"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="5"
      onClick={[Function]}
      type="button"
      value="5"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="6"
      onClick={[Function]}
      type="button"
      value="6"
    />
    <br />
  </Fragment>,
  <Fragment>
    <input
      className="operator"
      disabled={false}
      key=" - "
      onClick={[Function]}
      type="button"
      value=" - "
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="7"
      onClick={[Function]}
      type="button"
      value="7"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="8"
      onClick={[Function]}
      type="button"
      value="8"
    />
    <br />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="9"
      onClick={[Function]}
      type="button"
      value="9"
    />
  </Fragment>,
  <Fragment>
    <input
      className="operator"
      disabled={false}
      key=" * "
      onClick={[Function]}
      type="button"
      value=" * "
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="x+x"
      onClick={[Function]}
      type="button"
      value="x+x"
    />
    <br />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="x-x"
      onClick={[Function]}
      type="button"
      value="x-x"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="x*x"
      onClick={[Function]}
      type="button"
      value="x*x"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="x/x"
      onClick={[Function]}
      type="button"
      value="x/x"
    />
    <br />
  </Fragment>,
  <Fragment>
    <input
      className="clear"
      disabled={false}
      key="C"
      onClick={[Function]}
      type="button"
      value="C"
    />
  </Fragment>,
  <Fragment>
    <input
      disabled={false}
      key="0"
      onClick={[Function]}
      type="button"
      value="0"
    />
  </Fragment>,
  <Fragment>
    <input
      className="equal"
      disabled={false}
      key=" = "
      onClick={[Function]}
      type="button"
      value=" = "
    />
    <br />
  </Fragment>,
  <Fragment>
    <input
      className="operator"
      disabled={false}
      key=" / "
      onClick={[Function]}
      type="button"
      value=" / "
    />
  </Fragment>,
]
`;
